mailgun/resource_mailgun_credential.go

```diff
@@ -0,0 +1,217 @@
+package mailgun
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/mailgun/mailgun-go/v3"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type mailgunCredential struct {
```

Comment:
This struct is only for logging?

Reply:
Yes. Initially, I use for something else.